### PR TITLE
Replace heatmap with treemap on the memory page

### PR DESCRIPTION
**Pull request by peterdjlee**
Originally reviewed and approved in #2084 -- isolating code that replaced the heatmap with treemap on the memory page.
Screen Shot 2020-06-26 at 11 27 27 AM

### CODE REVIEW

File: packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart [hunk 1]
```diff
@@ -369,7 +366,7 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
       mainAxisAlignment: MainAxisAlignment.end,
       children: [
         Expanded(child: snapshotDisplay),
-        if (hasDetails) const SizedBox(width: defaultSpacing),
+        hasDetails ? const SizedBox(width: defaultSpacing) : const SizedBox(),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart:369]: why is this change necessary?

File: packages/devtools_app/lib/src/memory/memory_treemap.dart [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+        sizes = InstructionsSize.fromSnapshot(controller);
+      });
+    });
+
+    addAutoDisposeListener(controller.filterNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(rebuild: true);
+      });
+    });
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/memory/memory_treemap.dart:63]: so does filter work in this CL or not? If not, I don't think we should be listening to filterNotifier

File: packages/devtools_app/lib/src/memory/memory_treemap.dart [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    //   setState(() {
+    //     if (_trySelectItem()) {
+    //       closeAutoCompleteOverlay();
+    //     }
+    //   });
+    // });
+
+    addAutoDisposeListener(controller.searchAutoCompleteNotifier, () {
+      setState(autoCompleteOverlaySetState(controller, context));
+    });
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/memory/memory_treemap.dart:83]: similar comment here. If search is not hooked up, I don't think we should be listening for search autocomplete

File: packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart (hunk 1 above)
Comment (peterdjlee) [reply packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart]: I think initially it was because the line didn't fit in one line but it should fit in one line. Reverted.

File: packages/devtools_app/lib/src/memory/memory_treemap.dart (hunk 2 above)
Comment (peterdjlee) [reply packages/devtools_app/lib/src/memory/memory_treemap.dart]: Commented out

File: packages/devtools_app/lib/src/memory/memory_treemap.dart (hunk 3 above)
Comment (peterdjlee) [reply packages/devtools_app/lib/src/memory/memory_treemap.dart]: Commented out
